### PR TITLE
WIP - Parse script task content as python using ast

### PR DESCRIPTION
**Pull request by nat-n**
- validate that the script content is just a function call
- validate that only known arguments are referenced

Also
- reorganise related tests and fixtures
- reorganise filters

In progress
- cover a failing edge case or two
- revisit script task validation logic
- clean up

### CODE REVIEW

File: poethepoet/task/script.py [hunk 1]
```diff
@@ -38,26 +39,24 @@ def _handle_run(
     ) -> int:
         # TODO: check whether the project really does use src layout, and don't do
         #       sys.path.append('src') if it doesn't
-        target_module, target_callable, call_params = self._parse_content(self.content)
         named_args = self.parse_named_args(extra_args)
-        if named_args is not None:
-            if call_params is None:
-                call_params = f"**({named_args!r})"
-            else:
-                call_params = f"**({self._resolve_args(call_params, named_args)})"
```
Comment (nat-n) [inline poethepoet/task/script.py:47]: moving away from kwargs seems to change the handling of Optional arguments. More thought it required about how this should be handled.

Comment (cigani) [reply poethepoet/task/script.py:47]: It's tricky for sure... Maybe we should approach it as you suggested via a failing test and attempt some simple solutions? If I recall your primary concern was how can we properly handle positional arguments in this situation. 

Maybe we think of position as the "Simple" that should work by default and kwargs should be the complex which requires more user input?

Comment (nat-n) [reply poethepoet/task/script.py:47]: I don't see a way to make it work. if you specify a task like in one of the tests you wrote like:
```
script = "pkg:fun(optional=optional)"
```
then with the new approach some value much be given for the optional kwarg in the call to fun. If no argument value was provided then the value will be None, but either way the default value for that kwarg as given in the definition of `fun` will not be used.
